src/puppet-dom.js

```diff
@@ -14,7 +14,10 @@
     var onDataReady = options.callback;
     this.element = options.listenTo || document.body;
     var clickHandler = this.clickHandler.bind(this);
-    var historyHandler = this.historyHandler.bind(this);
+    this.historyHandler = this.historyHandler.bind(this);
+
+    /* no need to wait for data to be ready to handle callback */
+    this.element.addEventListener('puppet-redirect-pushstate', this.historyHandler);
```

Comment:
I can check. Do you mean `2.4.0` branch tests or `fix-redirect-handler` branch tests?